rust/lancedb/src/index.rs

```diff
@@ -177,6 +177,8 @@ pub struct IndexConfig {
     /// Currently this is always a Vec of size 1.  In the future there may
     /// be more columns to represent composite indices.
     pub columns: Vec<String>,
+    /// The version of the index
+    pub version: u64,
```

Comment:
What is an index's "version"? Where does that come from?

Is this derived from `dataset_version` or something else?

https://github.com/lancedb/lance/blob/b572905d22b1f168fccb2768c211642ec894bff3/rust/lance-table/src/format/index.rs#L26-L27

Reply:
yes, it's dataset_version